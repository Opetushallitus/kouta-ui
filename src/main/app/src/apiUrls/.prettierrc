{
  "printWidth": 1000,
  "trailingComma": "all",
  "tabWidth": 2,
  "semi": true,
  "singleQuote": true
}