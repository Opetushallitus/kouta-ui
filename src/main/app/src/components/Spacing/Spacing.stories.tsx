import React from 'react';

import { storiesOf } from '@storybook/react';

import Spacing from './index';

storiesOf('Spacing', module).add('Basic', () => (
  <>
    <Spacing marginTop={1} marginBottom={2} marginLeft={3} marginRight={4}>
      Margin
    </Spacing>
    <Spacing paddingTop={1} paddingBottom={2} paddingLeft={3} paddingRight={4}>
      Padding
    </Spacing>
  </>
));
