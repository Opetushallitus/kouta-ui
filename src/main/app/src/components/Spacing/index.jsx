import styled from 'styled-components';

import { spacingCss } from '../../theme';

const Spacing = styled.div`
  ${spacingCss};
`;

export default Spacing;
