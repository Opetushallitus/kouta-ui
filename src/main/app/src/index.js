import 'core-js/features/object/keys';
import 'core-js/features/object/values';
import 'core-js/features/promise';
import 'core-js/features/array/includes';
import 'core-js/features/array/every';
import 'core-js/features/array/flat-map';

import React from 'react';
import ReactDOM from 'react-dom';
import createBrowserHistory from 'history/createBrowserHistory';
import { urls as ophUrls } from 'oph-urls-js';
import get from 'lodash/get';
import merge from 'lodash/merge';

import * as serviceWorker from './serviceWorker';
import App from './components/App';
import createStore from './state';
import defaultTheme from './theme';
import configureUrls from './apiUrls';
import createLocalisation from './localisation';
import { getLocalisation } from './apiUtils';
import getTranslations from './translations';
import createHttpClient from './httpClient';

const history = createBrowserHistory({ basename: 'kouta' });

serviceWorker.unregister();

const isDev = process.env.NODE_ENV === 'development';

const loadLocalisation = async ({
  namespace,
  language,
  httpClient,
  apiUrls,
}) => {
  const localisation = await getLocalisation({
    category: namespace,
    locale: language,
    httpClient,
    apiUrls,
  });

  const translations = getTranslations();

  return get(translations, [language, namespace])
    ? merge({}, translations[language][namespace], localisation || {})
    : localisation;
};

(async () => {
  const apiUrls = await configureUrls(ophUrls);

  const httpClient = createHttpClient({
    apiUrls,
    callerId: !isDev ? process.env.REACT_APP_CALLER_ID : null,
  });

  const localisationInstance = createLocalisation({
    debug: isDev,
    loadLocalisation: ({ namespace, language }) =>
      loadLocalisation({ namespace, language, httpClient, apiUrls }),
  });

  window.__i18n__ = localisationInstance;

  const { store, persistor } = createStore({
    apiUrls,
    httpClient,
    history,
    localisation: localisationInstance,
  });

  window.__store__ = store;

  ReactDOM.render(
    <App
      store={store}
      theme={defaultTheme}
      urls={apiUrls}
      httpClient={httpClient}
      history={history}
      localisation={localisationInstance}
      persistor={persistor}
    />,
    document.getElementById('root'),
  );
})();
