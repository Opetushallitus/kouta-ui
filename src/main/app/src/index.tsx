import React from 'react';

import { Globals } from '@react-spring/web';
import { inspect } from '@xstate/inspect';
import { createBrowserHistory } from 'history';
import { urls as ophUrls } from 'oph-urls-js';
import ReactDOM from 'react-dom';

import createHttpClient from './httpClient';
import { createDefaultLocalization } from './localization';
import App from './pages/App';
import { createStore } from './state';
import defaultTheme from './theme';
import { configure as configureUrls } from './urls';
import { isPlaywright } from './utils';

Globals.assign({
  skipAnimation: isPlaywright,
});

if (import.meta.env.VITE_XSTATE_INSPECTOR) {
  inspect({
    iframe: false, // open in new window
  });
}

const history = createBrowserHistory({ basename: 'kouta' });

(async () => {
  let apiUrls = ophUrls;

  const httpClient = createHttpClient({
    apiUrls,
    callerId: import.meta.env.VITE_CALLER_ID,
  });

  apiUrls = await configureUrls(apiUrls, httpClient);

  const localizationInstance = await createDefaultLocalization({
    httpClient,
    apiUrls,
  });

  const { store, persistor } = createStore();

  ReactDOM.render(
    <App
      store={store}
      theme={defaultTheme}
      urls={apiUrls}
      httpClient={httpClient}
      history={history}
      localization={localizationInstance}
      persistor={persistor}
    />,
    document.getElementById('root')
  );
})();
