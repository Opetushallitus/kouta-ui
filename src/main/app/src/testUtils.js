import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import defaultTheme from './theme';

const makeFnWithTheme = fn => tree => {
  return fn(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>);
};

export const shallowWithTheme = makeFnWithTheme(shallow);

export const mountWithTheme = makeFnWithTheme(mount);

export const setMockDate = date => {
  const _Date = Date;

  global.Date.now = () => +date;

  return () => {
    global.Date = _Date;
  };
};
