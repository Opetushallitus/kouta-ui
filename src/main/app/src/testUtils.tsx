import React, { PropsWithChildren } from 'react';

import { configureStore } from '@reduxjs/toolkit';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

import { createRootReducer } from './state/rootReducer';
import defaultTheme from './theme';

const makeFnWithTheme = fn => tree => {
  return fn(<ThemeProvider theme={defaultTheme}>{tree}</ThemeProvider>);
};

export const renderWithTheme = makeFnWithTheme(render);

export const mountWithTheme = tree => {
  const { container } = makeFnWithTheme(render)(tree);
  return container.firstChild;
};

export const setMockDate = date => {
  const _Date = Date;

  global.Date.now = () => Number(date);

  return () => {
    global.Date = _Date;
  };
};

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: createRootReducer({}), preloadedState }),
    ...renderOptions
  }
) {
  // eslint-disable-next-line @typescript-eslint/ban-types
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return (
      <Provider store={store}>
        <ThemeProvider theme={defaultTheme}>{children}</ThemeProvider>
      </Provider>
    );
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';
