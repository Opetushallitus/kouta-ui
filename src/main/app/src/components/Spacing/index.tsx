import styled from 'styled-components';

import { space } from '#/src/system';

export const Spacing = styled.div`
  ${space}
`;

export default Spacing;
