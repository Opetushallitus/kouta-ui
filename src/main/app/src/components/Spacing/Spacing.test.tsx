import React from 'react';

import Spacing from './index';
import { mountWithTheme } from '#/src/testUtils';

test('renders correctly', () => {
  expect(
    mountWithTheme(
      <>
        <Spacing marginLeft={1} marginRight={2} marginBottom={3} marginTop={4}>
          With margins
        </Spacing>
        <Spacing
          paddingLeft={1}
          paddingRight={2}
          paddingBottom={3}
          paddingTop={4}
        >
          With paddings
        </Spacing>
      </>
    )
  ).toMatchSnapshot();
});
