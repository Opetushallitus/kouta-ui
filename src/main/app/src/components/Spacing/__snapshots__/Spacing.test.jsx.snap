// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  .c0 {
  margin-left: 8px;
  margin-right: 16px;
  margin-bottom: 24px;
  margin-top: 32px;
}

<Spacing
    marginBottom={3}
    marginLeft={1}
    marginRight={2}
    marginTop={4}
  >
    <div
      className="c0"
    >
      With margins
    </div>
  </Spacing>,
  .c0 {
  padding-left: 8px;
  padding-right: 16px;
  padding-bottom: 24px;
  padding-top: 32px;
}

<Spacing
    paddingBottom={3}
    paddingLeft={1}
    paddingRight={2}
    paddingTop={4}
  >
    <div
      className="c0"
    >
      With paddings
    </div>
  </Spacing>,
]
`;
